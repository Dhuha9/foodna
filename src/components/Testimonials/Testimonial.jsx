import React from 'react';
import PropTypes from 'prop-types';
import './Testimonial.scss';

export default function Testimonial({ text, personImage, name, profession }) {
  return (
    <>
      <blockquote className="p-4 blockquote">
        <h4>
          <q>{text}</q>
        </h4>
      </blockquote>
      <div className="d-block m-auto w-25 h-25">
        <img
          className="d-block m-auto w-50 h-50 testimonial-person"
          src={personImage}
          alt="the person"
        />
      </div>
      <div className="mt-3 mb-5">
        <figcaption className="blockquote-footer">
          &mdash; {name + ', ' + profession}
        </figcaption>
      </div>
    </>
  );
}

Testimonial.propTypes = {
  text: PropTypes.string,
  personImage: PropTypes.string,
  name: PropTypes.string,
  profession: PropTypes.string,
};
