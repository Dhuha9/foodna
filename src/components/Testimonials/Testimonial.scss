.testimonial-person {
  border-radius: 50%;
}

q {
  font-family: Roboto;
  font-size: 32px;
  font-style: normal;
  font-weight: 400;
  line-height: 49px;
  letter-spacing: 0em;
  text-align: center;
}

.blockquote-footer {
  color: #03b664;
  font-family: Roboto;
  font-size: 18px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0em;
  text-align: center;
}
