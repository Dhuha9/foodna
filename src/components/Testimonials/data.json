[
  {
    "id": 0,
    "text": "A well-known quote, contained in a blockquote element",
    "personImage": "TA",
    "name": "Tabarek Ayad",
    "profession": "Developer"
  },
  {
    "id": 1,
    "text": "Another well-known quote, contained in a blockquote element",
    "personImage": "TA1",
    "name": "TA",
    "profession": "Developer"
  }
]
