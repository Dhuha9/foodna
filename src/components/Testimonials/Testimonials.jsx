import React from 'react';
import Testimonial from './Testimonial';
import data from './data.json';
import { Carousel } from 'react-bootstrap';
import TA from '../../dist/img/TA.jpg';
import TA1 from '../../dist/img/TA1.jpg';

const imgMap = { TA: TA, TA1: TA1 };

function Testimonials() {
  return (
    <div className="w-50 m-auto">
      <Carousel>
        {data.map((testimonialsData, index) => (
          <Carousel.Item key={index} className="mt-3">
            <Testimonial
              {...testimonialsData}
              personImage={imgMap[testimonialsData.personImage]}
            />
          </Carousel.Item>
        ))}
      </Carousel>
    </div>
  );
}

export default Testimonials;
