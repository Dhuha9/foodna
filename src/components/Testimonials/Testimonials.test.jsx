import React from 'react';
import renderer from 'react-test-renderer';
import Testimonials from './Testimonials';

it('Testimonials snapshot', () => {
  const tree = renderer.create(<Testimonials />).toJSON();
  expect(tree).toMatchSnapshot();
});
