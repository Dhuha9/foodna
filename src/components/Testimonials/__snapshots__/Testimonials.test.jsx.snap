// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials snapshot 1`] = `
<div
  className="w-50 m-auto"
>
  <div
    className="carousel slide"
    onKeyDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="mt-3 active carousel-item"
      >
        <blockquote
          className="p-4 blockquote"
        >
          <h4>
            <q>
              A well-known quote, contained in a blockquote element
            </q>
          </h4>
        </blockquote>
        <div
          className="d-block m-auto w-25 h-25"
        >
          <img
            alt="the person"
            className="d-block m-auto w-50 h-50 testimonial-person"
            src="TA.jpg"
          />
        </div>
        <div
          className="mt-3 mb-5"
        >
          <figcaption
            className="blockquote-footer"
          >
            — 
            Tabarek Ayad, Developer
          </figcaption>
        </div>
      </div>
      <div
        className="mt-3 carousel-item"
      >
        <blockquote
          className="p-4 blockquote"
        >
          <h4>
            <q>
              Another well-known quote, contained in a blockquote element
            </q>
          </h4>
        </blockquote>
        <div
          className="d-block m-auto w-25 h-25"
        >
          <img
            alt="the person"
            className="d-block m-auto w-50 h-50 testimonial-person"
            src="TA1.jpg"
          />
        </div>
        <div
          className="mt-3 mb-5"
        >
          <figcaption
            className="blockquote-footer"
          >
            — 
            TA, Developer
          </figcaption>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
</div>
`;
